src/lib/uri/uri.h

```diff
@@ -33,14 +33,12 @@
 
 #include <stddef.h>
 #include <stdbool.h>
-#include <netdb.h> /* NI_MAXHOST, NI_MAXSERV */
-#include <limits.h> /* _POSIX_PATH_MAX */
 
 #if defined(__cplusplus)
 extern "C" {
 #endif /* defined(__cplusplus) */
 
-struct uri {
+struct internal_uri_impl {
```

Comment:
Fixed all. I add uri_constants.h to prevent including uri.h in parser.rl. Because uri.h deals with `struct uri`, which is public instead of parser.rl which deals with uri_raw. But i think that such little header is excess.